projects/packages/waf/tests/php/test-waf-runner.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Standalone bootstrap test suite.
+ *
+ * @package automattic/jetpack-waf
+ */
+
+use Automattic\Jetpack\Waf\Waf_Runner;
+
+/**
+ * Runtime test suite.
+ */
+final class WafRunnerTest extends PHPUnit\Framework\TestCase {
```

Comment:
Several issues there - first is that I'm not sure how beneficial such a test would be? With us mocking the filesystem so heavily, I can't imagine any reasonable scenario where it wouldn't pass. The best approach to testing for this would likely be an integration test imo. The second reason is how many issues I ran into trying to get tests to work. After discussing it with Benedict, we decided that the best path forward would be to push this PR so it could be reviewed, and then you an I can sync to discuss further testing if you have suggestions.

Reply:
> I'm not sure how beneficial such a test would be?

Well, the title of this task is/was "Generate a rules.php file", and currently we are not testing at all whether that actually happens. So I'd argue it is very beneficial. And btw I am fine if we only had an integration test, but currently we don't have any at all.

Having said that, I am totally fine with adding the/more tests later on especially since you've already discussed the matter. 👌🏻